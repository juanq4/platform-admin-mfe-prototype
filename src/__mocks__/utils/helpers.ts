import { isEmpty } from "@q4/nimbus-ui";

export function mockWindowMethods(): Window {
  return Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
}

export function mockClipboardWrite(): void {
  const orginalClipboard = { ...navigator.clipboard };

  beforeAll(() => {
    if (!isEmpty(navigator.clipboard)) return;

    Object.defineProperty(navigator, "clipboard", {
      value: {
        writeText: () => Promise.resolve(),
      },
    });
  });

  afterAll(() => {
    Object.assign(navigator.clipboard, {
      ...orginalClipboard,
    });
  });
}
