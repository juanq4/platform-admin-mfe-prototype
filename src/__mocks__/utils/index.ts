export * from "./helpers";
export * from "./permissions";
export * from "./wrappers";
