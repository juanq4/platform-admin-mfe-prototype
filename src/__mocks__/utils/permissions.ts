import { Permission } from "@q4/platform-definitions";

export const MockQ4AdminPermissions = [
  Permission.ReadUsers,
  Permission.ManageUsers,
  Permission.ReadOrgs,
  Permission.ManageOrgs,
];

export const MockQ4SupportPermissions = [Permission.ReadUsers, Permission.ReadOrgs, Permission.ManageOrgs];

export const MockCorporateAdminPermissions = [Permission.ReadUsers, Permission.ManageUsers];

export const MockCorporateSupportPermissions: Permission[] = [];
