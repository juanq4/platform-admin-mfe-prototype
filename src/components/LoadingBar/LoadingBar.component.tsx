import { LoadingBarContainer, Bar, LoadingContainer, LoadingText } from "./LoadingBar.style";

interface LoadingBarProps {
  message?: string;
  className?: string;
}

// TODO: Replace this with the LoadingBar from new UI libary once ready
export const LoadingBar = (props: LoadingBarProps): JSX.Element => {
  const { message, className } = props;
  return (
    <LoadingContainer className={className}>
      {message && <LoadingText>{message}</LoadingText>}

      <LoadingBarContainer>
        <Bar />
      </LoadingBarContainer>
    </LoadingContainer>
  );
};
