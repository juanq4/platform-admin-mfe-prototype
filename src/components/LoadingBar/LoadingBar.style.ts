import { Typography } from "@mui/material";
import { keyframes, styled } from "@mui/material/styles";

export const LoadingContainer = styled("div")`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  position: fixed;
  width: 100vw;
  z-index: 10;
  background: white;
  gap: 16px;
`;

export const LoadingText = styled(Typography)`
  color: #1f2329;
  text-align: center;
  font-family: Inter;
  font-size: 20px;
  font-style: normal;
  font-weight: 600;
  line-height: 28px;
`;

export const LoadingAnimation = keyframes`
 0% {
    left: 0;
    width: 0;
  }

  25% {
    left: 0;
    width: 33%;
  }

  50% {
    left: 15%;
    width: 50%;
  }

  75% {
    left: 30%;
    width: 70%;
  }

  100% {
    left: 100%;
    width: 0;
  }
`;

export const LoadingBarContainer = styled("div")`
  background: #d3d8de;
  margin: 10px auto;
  border-radius: 16px;
  max-width: 80vw;

  @media screen and (min-width: 1025px) {
    width: 680px;
  }

  @media screen and (min-width: 769px) and (max-width: 1024px) {
    width: 728px;
  }

  @media screen and (min-width: 429px) and (max-width: 768px) {
    width: 550px;
  }

  @media screen and (max-width: 428px) {
    width: 288px;
  }
`;

export const Bar = styled("div")`
  height: 4px;
  border-radius: 16px;
  position: relative;
  background: linear-gradient(90deg, #d71598 0%, #a513ce 49.43%, #284eed 100%);
  width: 0%;
  left: 0%;
  animation: ${LoadingAnimation} 3s linear infinite;
`;
