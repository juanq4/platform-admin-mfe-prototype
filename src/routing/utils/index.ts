export * from "./api";
export * from "./s3checks";
export * from "./versionManager";
