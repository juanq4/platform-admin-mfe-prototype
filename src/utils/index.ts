export * from "./account";
export * from "./api";
export * from "./entitlement";
export * from "./error";
export * from "./feature";
export * from "./logout";
export * from "./organization";
export * from "./permission";
export * from "./role";
export * from "./route";
export * from "./subscription";
export * from "./user";
export * from "./environment";
