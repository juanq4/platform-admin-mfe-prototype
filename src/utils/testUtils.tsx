/* eslint-disable import/export */
import { MockedProvider as ApolloMockedProvider } from "@apollo/client/testing";
import { ThemeProvider } from "@emotion/react";
import { StyleGuide, ConfigProvider as NimbusConfig } from "@q4/nimbus-ui";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import theme from "../styles/theme";

// https://testing-library.com/docs/react-testing-library/setup/
const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
  styleGuide = StyleGuide.V2,
): RenderResult =>
  render(ui, {
    // Common middleware to all tests
    wrapper: ({ children }) => (
      <NimbusConfig styleGuide={styleGuide}>
        <ApolloMockedProvider>
          <ThemeProvider theme={theme}>{children}</ThemeProvider>
        </ApolloMockedProvider>
      </NimbusConfig>
    ),
    ...options,
  });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
