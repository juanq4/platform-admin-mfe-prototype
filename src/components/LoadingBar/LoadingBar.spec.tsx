import React from "react";
import { render, screen } from "../../utils/testUtils";
import { LoadingBar } from "../LoadingBar/LoadingBar.component";

const mockMesage = "I'm a message";

describe("LoadingBar", () => {
  test("expect to render", () => {
    const { container } = render(<LoadingBar />);

    expect(container).toMatchSnapshot();
  });

  test("11519452: [Given] a message is provided [Then] expect message to show on loader", () => {
    render(<LoadingBar message={mockMesage} />);

    expect(screen.getByText(mockMesage)).toBeInTheDocument();
  });

  test("11519453: [Given] a message is not provided [Then] expect no message to be show on loader", () => {
    render(<LoadingBar />);

    expect(screen.queryByText(mockMesage)).not.toBeInTheDocument();
  });
});
