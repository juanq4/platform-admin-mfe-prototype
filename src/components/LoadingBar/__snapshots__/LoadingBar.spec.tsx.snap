// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingBar expect to render 1`] = `
@keyframes animation-0 {
  0% {
    left: 0;
    width: 0;
  }

  25% {
    left: 0;
    width: 33%;
  }

  50% {
    left: 15%;
    width: 50%;
  }

  75% {
    left: 30%;
    width: 70%;
  }

  100% {
    left: 100%;
    width: 0;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 100vh;
  position: fixed;
  width: 100vw;
  z-index: 10;
  background: white;
  gap: 16px;
}

.emotion-1 {
  background: #d3d8de;
  margin: 10px auto;
  border-radius: 16px;
  max-width: 80vw;
}

@media screen and (min-width: 1025px) {
  .emotion-1 {
    width: 680px;
  }
}

@media screen and (min-width: 769px) and (max-width: 1024px) {
  .emotion-1 {
    width: 728px;
  }
}

@media screen and (min-width: 429px) and (max-width: 768px) {
  .emotion-1 {
    width: 550px;
  }
}

@media screen and (max-width: 428px) {
  .emotion-1 {
    width: 288px;
  }
}

.emotion-2 {
  height: 4px;
  border-radius: 16px;
  position: relative;
  background: linear-gradient(90deg, #d71598 0%, #a513ce 49.43%, #284eed 100%);
  width: 0%;
  left: 0%;
  -webkit-animation: animation-0 3s linear infinite;
  animation: animation-0 3s linear infinite;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      />
    </div>
  </div>
</div>
`;
